test/studies/adventOfCode/2022/day11/bradc/day11-par.chpl

```diff
@@ -0,0 +1,181 @@
+use IO, List, Barriers;
+
+config const numRounds = 20;
+
+// things that define a monkey
+class Monkey {
+  var id: int,                 // an integer ID
+      Items: [0..1] list(int, parSafe=true),  // two lists of its items
+      current = 0, next=1,     // which list is the current vs. next one
+      op: owned MathOp,            // the operator the monkey does
+      divisor: int,            // the divisor for its % operation check
+      targetMonkey: 2*int,     // the target monkeys it throws items to
+      numInspected: int;       // the count of how many inspections we've done
+}
+
+/*
+var m1 = new Monkey();
+var m2 = new Monkey();
+var m3 = new Monkey();
+var m4 = new Monkey();
+*/
+
+// An array of monkeys
+var Monkeys = readMonkeys();
+//writeln(Monkeys);
+
+
+// This tells whether a given monkey can proceed when it is out of items.
+// Initially, only monkey 0 can since nobody can throw items into its
+// list for this round, only for subsequent rounds.
+var canPause: sync int = 0;
+
+
+// Our main parallel simulation loop for the monkeys
+var bar = new Barrier(Monkeys.size);
+coforall monkey in Monkeys {
+  for r in 1..numRounds {
+    // Process any items that are in our list of current items
+    monkey.processItems(canPause);
+
+    // Make sure everyone is done so that we don't start changing our
+    // items list for the next round while monkeys are still adding
+    // things to it.
+    bar.barrier();
+
+    // Reset our items lists so that the list for the next round
+    // bcomes the new one
+    monkey.resetForNextRound(canPause);
+
+    // Make sure everyone has swapped their item lists so that we
+    // don't start throwing things into their next list of items
+    // before they're done swapping it for this round
+    bar.barrier();
+  }
+}
+writeln(Monkeys.numInspected);
+
+
+// Here's how one monkey processes its items
+proc Monkey.processItems(canPause) {
+  // subtle MCM issue here if these two expressions are swapped... woof
+  while (canPause.readXX() != id || Items[current].size > 0) {
+    while Items[current].size > 0 {
```

Comment:
I'd consider creating a method called `currentItem` for this/

Reply:
I'm not following:  for which part?

Reply:
For `Items[current]`. Just a nice shorthand, in my opinion, though possible with some overhead unless it's made `inline` (but we don't wan to teach inline)